src/storage/src/sink/kafka.rs

```diff
@@ -1058,21 +1113,57 @@ where
     builder.build(move |_capabilities| async move {
```

Comment:
Re: our discussion in chat: IIUC we could just initialize the healthchecker here, without needing to inspect the `sink_state`... it's guaranteed to be `Init` here. It looks like with the new setup we don't need the state machine at all, which is great.

Something for a future PR, though.